addons/account/models/account_move.py

```diff
@@ -2924,10 +2924,13 @@ def _get_fields_to_copy_recurring_entries(self, values):
     def _prepare_invoice_aggregated_taxes(self, filter_invl_to_apply=None, filter_tax_values_to_apply=None, grouping_key_generator=None):
         self.ensure_one()
 
-        base_lines = [
-            x._convert_to_tax_base_line_dict()
-            for x in self.line_ids.filtered(lambda x: x.display_type == 'product' and (not filter_invl_to_apply or filter_invl_to_apply(x)))
-        ]
+        base_lines = []
+        is_filter_invl_applied = False
+        for line in self.line_ids.filtered(lambda x: x.display_type == 'product'):
+            if not filter_invl_to_apply or filter_invl_to_apply(line):
```

Comment:
`filter_invl_to_apply` doesn't seem to be used in SA, I don't get why it would cause and issue in that country :thinking:

Reply:
You are partially right, the call is done in `account_edi_ubl_cii` module
https://github.com/odoo/odoo/blob/d425795ee00815602931135bb83633d202a4ac3b/addons/account_edi_ubl_cii/models/account_edi_xml_ubl_20.py#L452-L454
But it will use the filter method in `l10n_sa_edi`
https://github.com/odoo/odoo/blob/d425795ee00815602931135bb83633d202a4ac3b/addons/l10n_sa_edi/models/account_edi_xml_ubl_21_zatca.py#L201-L206